### DISCUSSION TITLE
Prepared Statement builders

### DISCUSSION THREAD

**Discussion by tychobrailleur** [ho-dev/HattrickOrganizer#1968] [Dev]
I am doing some refactoring on the database side of things, for the sake of adding unit tests & integration tests, and I had a question around prepared statement builders (i.e. `DBManager.PreparedStatementBuilder`): my understanding of their primary intent is to cache them so that they can be reused.  Is that a fair thing to say? 

If that's the case, I am wondering of making this easier to manage, and possibly also looking at ways to cache the data rather than retrieving directly from the DB.  That's something I am currently investigating.

**Comment by wsbrenk** [comment]
I'm not the big SQL expert. But as I understand it, the prepared statements have a performance advantage when they are often reused since the parsing step is only done one time.

As far as I understand is the hsqldb already a kind of in memory database - so doubling the data into caches won't give a great benefit but requires a lot of extra memory space!?
